From ce2ae1735b3c63fb0947a020a1e058f8789bae70 Mon Sep 17 00:00:00 2001
From: Henri Yandell <bayard@apache.org>
Date: Tue, 21 Aug 2007 16:41:16 +0000
Subject: [PATCH] Applying the patch from DIGESTER-114 - stopping an
 IllegalArgumentException when the element has no attributes. After reviewing
 test and fix, it all looks good to me.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/digester/trunk@568192 13f79535-47bb-0310-9956-ffa450edef68
---
 .../apache/commons/digester/SetPropertyRule.java  |  4 ++++
 .../commons/digester/SetPropertyRuleTestCase.java | 15 +++++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/src/java/org/apache/commons/digester/SetPropertyRule.java b/src/java/org/apache/commons/digester/SetPropertyRule.java
index 092d7f705..08cff5930 100644
--- a/src/java/org/apache/commons/digester/SetPropertyRule.java
+++ b/src/java/org/apache/commons/digester/SetPropertyRule.java
@@ -103,6 +103,10 @@ public SetPropertyRule(String name, String value) {
      */
     public void begin(Attributes attributes) throws Exception {
 
+        if (attributes.getLength() == 0 ) {
+            return;
+        }
+
         // Identify the actual property name and value to be used
         String actualName = null;
         String actualValue = null;
diff --git a/src/test/org/apache/commons/digester/SetPropertyRuleTestCase.java b/src/test/org/apache/commons/digester/SetPropertyRuleTestCase.java
index d7af0af3a..5f0a89105 100644
--- a/src/test/org/apache/commons/digester/SetPropertyRuleTestCase.java
+++ b/src/test/org/apache/commons/digester/SetPropertyRuleTestCase.java
@@ -201,6 +201,21 @@ private Reader xmlTestReader(String xml) throws IOException {
         return new StringReader(xml);
     }
 
+    /**
+     * Test SetPropertyRule when matched XML element has no attributes.
+     * See: DIGESTER-114
+     */
+    public void testElementWithNoAttributes() throws Exception {
+        String TEST_XML_3 = "<?xml version='1.0'?><root><set/></root>";
+
+        // Set up the rules we need
+        digester.addObjectCreate("root", "org.apache.commons.digester.SimpleTestBean");
+        digester.addSetProperty("root/set", "name", "value");
+
+        // Parse the input - should not throw an exception
+        SimpleTestBean bean = (SimpleTestBean) digester.parse(xmlTestReader(TEST_XML_3));
+    } 
+
 }
 
 
